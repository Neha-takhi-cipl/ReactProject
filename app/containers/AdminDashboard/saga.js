import { call, put, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';
import { API_ENDPOINTS } from 'appConfig';
import auth from 'utils/auth';
import {
  GET_JOBS_REQUEST,
  ADD_AGENT_REQUEST,
  UPDATE_AGENT_REQUEST,
  GET_AGENTS_REQUEST,
  UPLOAD_PICTURE_REQUEST,
} from './constants';
import {
  getJobsSuccess,
  getJobsError,
  addAgentSuccess,
  addAgentError,
  updateAgentError,
  updateAgentSuccess,
  getAgentsSuccess,
  getAgentsError,
  uploadPictureSuccess,
  uploadPictureError,
} from './actions';

/**
 * @description SignUp the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* getJobs() {
  const requestURL = `${API_ENDPOINTS.adminJobs}`;
  const requestParam = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(getJobsSuccess(response));
  } catch (err) {
    yield put(getJobsError(err));
  }
}

/**
 * @description Create a new agent request/response handler
 * @author PravinKumar
 * @since 23 Jun 2018
 */
export function* addAgent(action) {
  const requestURL = `${API_ENDPOINTS.agentRegister}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify(action.data),
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(addAgentSuccess(response));
  } catch (err) {
    yield put(addAgentError(err));
  }
}

/**
 * @description Update Agent request/response handler
 * @author PravinKumar
 * @since 23 Jun 2018
 */
export function* updateAgent(action) {
  const requestData = action.data;
  const requestURL = `${API_ENDPOINTS.updateAgent}/${action.id}`;
  if (!(!requestData.password && requestData.password !== '')) {
    delete requestData.password;
  }
  const requestParam = {
    method: 'PUT',
    body: JSON.stringify(requestData),
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(updateAgentSuccess(response));
  } catch (err) {
    yield put(updateAgentError(err));
  }
}

/**
 * @description get agent listings request/response handler
 * @author PravinKumar
 * @since 23 Jun 2018
 */
export function* getAgents() {
  const requestURL = `${API_ENDPOINTS.getAgents}`;
  const requestParam = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(getAgentsSuccess(response));
  } catch (err) {
    yield put(getAgentsError(err));
  }
}

/**
 * @description upload Picture  request/response handler
 * @author PravinKumar
 * @since 23 Jun 2018
 */
export function* uploadPicture(action) {
  const requestURL = `${API_ENDPOINTS.uploadPicture}`;
  const fd = new FormData();
  fd.append('picture', action.data);
  const requestParam = {
    method: 'POST',
    headers: {
      'x-access-token': auth.get('jwtToken'),
    },
    body: fd,
  };
  try {
    // Call our request helper (see 'utils/request')
    const response = yield call(request, requestURL, requestParam);
    yield put(uploadPictureSuccess(response));
  } catch (err) {
    yield put(uploadPictureError(err));
  }
}

/**
 * @description Root saga manages watcher lifecycle
 * @author PravinKumar
 * @since 17 May 2018
 */

export default function* Watcher() {
  yield takeLatest(GET_JOBS_REQUEST, getJobs);
  yield takeLatest(ADD_AGENT_REQUEST, addAgent);
  yield takeLatest(UPDATE_AGENT_REQUEST, updateAgent);
  yield takeLatest(GET_AGENTS_REQUEST, getAgents);
  yield takeLatest(UPLOAD_PICTURE_REQUEST, uploadPicture);
}
