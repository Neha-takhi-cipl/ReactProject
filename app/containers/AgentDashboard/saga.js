import { call, put, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';
import { API_ENDPOINTS } from 'appConfig';
import auth from 'utils/auth';
import {
  GET_NEW_JOBS_REQUEST,
  ASSIGN_JOBS_REQUEST,
} from './constants';
import {
  getNewJobsSuccess,
  getNewJobsError,
  assignJobToMeSuccess,
  assignJobToMeError,
} from './actions';

/**
 * @description get New Jobs  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* getNewJobs() {
  const requestURL = `${API_ENDPOINTS.agentJob}`;
  const requestParam = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(getNewJobsSuccess(response));
  } catch (err) {
    yield put(getNewJobsError(err));
  }
}

/**
 * @description assignJob to agent request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* assignJobToMe(action) {
  const jobId = action.data;
  const requestURL = `${API_ENDPOINTS.assignJob}/${jobId}`;
  const requestParam = {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(assignJobToMeSuccess(response));
  } catch (err) {
    yield put(assignJobToMeError(err));
  }
}

/**
 * @description Root saga manages watcher lifecycle
 * @author PravinKumar
 * @since 17 May 2018
 */

export default function* Watcher() {
  yield takeLatest(GET_NEW_JOBS_REQUEST, getNewJobs);
  yield takeLatest(ASSIGN_JOBS_REQUEST, assignJobToMe);
}
