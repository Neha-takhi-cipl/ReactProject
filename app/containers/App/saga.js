import { call, takeLatest, put } from 'redux-saga/effects';
import {
  ADD_USER_TO_STORE,
  INVITED_TRIP,
  GET_USER_REQUEST,
  TRIP_BY_URL,
} from 'containers/App/constants';
import {
  addUserToStoreSuccess,
  addUserToStoreError,
  invitedTripSuccess,
  invitedTripError,
  getUserDataSuccess,
  getUserDataError,
  tripByUrlSuccess,
  tripByUrlError,
} from 'containers/App/actions';
import auth from 'utils/auth';
import request from 'utils/request';
import { API_ENDPOINTS } from 'appConfig';
// import { makeSelectEmail, makeSelectPassword, makeSelectFacebookData } from 'containers/Login/selectors';


/**
 * @description facebookLogin the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* addUsertoStore(action) {
  try {
    yield put(addUserToStoreSuccess(action.data));
  } catch (err) {
    yield put(addUserToStoreError(err));
  }
}

/**
 * @description Invite Trip the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* inviteTrip(action) {
  try {
    yield put(invitedTripSuccess(action.data));
  } catch (err) {
    yield put(invitedTripError(err));
  }
}
/**
 * @description  Trip by url  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* tripByUrl(action) {
  try {
    yield put(tripByUrlSuccess(action.data));
  } catch (err) {
    yield put(tripByUrlError(err));
  }
}
/**
 * @description SignUp the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* getUser() {
  const requestURL = `${API_ENDPOINTS.getUserProfile}`;
  const requestParam = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'x-access-token': auth.get('jwtToken'),
    },
  };
  try {
    const response = yield call(request, requestURL, requestParam);
    yield put(getUserDataSuccess(response));
  } catch (err) {
    yield put(getUserDataError(err));
  }
}

// Individual exports for testing
export default function* defaultSaga() {
  yield takeLatest(ADD_USER_TO_STORE, addUsertoStore);
  yield takeLatest(INVITED_TRIP, inviteTrip);
  yield takeLatest(TRIP_BY_URL, tripByUrl);
  yield takeLatest(GET_USER_REQUEST, getUser);
}
