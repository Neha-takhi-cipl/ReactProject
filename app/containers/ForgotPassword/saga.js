import { call, put, select, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';
import { API_ENDPOINTS } from 'appConfig';
import { FORGOT_PASSWORD_REQUEST } from './constants';
import { makeSelectEmail } from './selectors';
import { forgetPasswordSuccess, forgetPasswordError } from './actions';

/**
 * @description forgetPassword the user  request/response handler
 * @author PravinKumar
 * @since 18 May 2018
 */
export function* forgetPassword() {
  // Select username from store
  const email = yield select(makeSelectEmail());
  const requestURL = `${API_ENDPOINTS.forgotPassword}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({ email }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    // Call our request helper (see 'utils/request')
    const Response = yield call(request, requestURL, requestParam);
    if (Response.status === 200) {
      yield put(forgetPasswordSuccess(Response));
    } else {
      yield put(forgetPasswordError(Response));
    }
  } catch (err) {
    yield put(forgetPasswordError(err));
  }
}

/**
 * @description SignUp the user  request/response handler
 * @author PravinKumar
 * @since 18 May 2018
 */
export default function* defaultSaga() {
  yield takeLatest(FORGOT_PASSWORD_REQUEST, forgetPassword);
}
