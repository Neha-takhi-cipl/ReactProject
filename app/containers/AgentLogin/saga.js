import { call, put, select, takeLatest } from 'redux-saga/effects';
import { LOGIN_REQUEST } from 'containers/App/constants';
import { loginSuccess, loginError } from 'containers/App/actions';
import request from 'utils/request';
import { makeSelectEmail, makeSelectPassword } from 'containers/AgentLogin/selectors';
import { BASE_URL } from '../../appConfig';

/**
 * @description login the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* login() {
  // Select username from store
  const email = yield select(makeSelectEmail());
  const password = yield select(makeSelectPassword());
  const requestURL = `${BASE_URL}api/login`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({
      email,
      password,
      type: 'agent',
    }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    const loginResponse = yield call(request, requestURL, requestParam);
    if (loginResponse.status === 200) {
      yield put(loginSuccess(loginResponse));
    } else {
      yield put(loginError(loginResponse));
    }
  } catch (err) {
    yield put(loginError(err));
  }
}

// Individual exports for testing
export default function* defaultSaga() {
  yield takeLatest(LOGIN_REQUEST, login);
}
