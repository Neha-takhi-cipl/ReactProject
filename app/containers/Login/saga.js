import { all, call, takeLatest, select, put } from 'redux-saga/effects';
import { FACEBOOK_LOGIN_REQUEST, LOGIN_REQUEST } from 'containers/App/constants';
import { loginSuccess, loginError, addUserToStoreSuccess } from 'containers/App/actions';
import request from 'utils/request';
import { history } from 'app';
import auth from 'utils/auth';
import { USER_TYPE, API_ENDPOINTS, AFTER_LOGIN_REDIRECT } from 'appConfig';
import { makeSelectEmail, makeSelectPassword, makeSelectFacebookData } from 'containers/Login/selectors';


/**
 * @description login the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* login() {
  // Select username from store
  const email = yield select(makeSelectEmail());
  const password = yield select(makeSelectPassword());
  const requestURL = `${API_ENDPOINTS.login}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({
      email,
      password,
      type: 'traveller',
    }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    const Response = yield call(request, requestURL, requestParam);
    if (Response.status === 200) {
      yield put(loginSuccess(Response));
      yield put(addUserToStoreSuccess(Response));
      yield all([
        call(auth.setToken, Response.data.token, true),
        call(auth.setUserInfo, Response.data, true),
      ]);
      if (Response.data.userType === USER_TYPE.traveller) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.traveller);
      } else if (Response.data.userType === USER_TYPE.agent) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.agent);
      } else if (Response.data.userType === USER_TYPE.admin) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.admin);
      }
    } else {
      yield put(loginError(Response));
    }
  } catch (err) {
    yield put(loginError(err));
  }
}

/**
 * @description facebookLogin the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* facebookLogin() {
  // Select username from store
  const facebookData = yield select(makeSelectFacebookData());
  const requestURL = `${API_ENDPOINTS.register}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({
      fullName: facebookData.name,
      email: facebookData.email,
      facebookId: facebookData.facebookId,
      picture: facebookData.picture,
      facebookLogin: true,
      password: false,
      userType: 'traveller',
    }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    // Call our request helper (see 'utils/request')
    const Response = yield call(request, requestURL, requestParam);
    yield put(loginSuccess(Response));
    yield put(addUserToStoreSuccess(Response));
    yield all([
      call(auth.setToken, Response.data.token, true),
      call(auth.setUserInfo, Response.data, true),
    ]);
    if (Response.data.userType === USER_TYPE.traveller) {
      yield call(forwardTo, AFTER_LOGIN_REDIRECT.traveller);
    } else if (Response.data.userType === USER_TYPE.agent) {
      yield call(forwardTo, AFTER_LOGIN_REDIRECT.agent);
    } else if (Response.data.userType === USER_TYPE.admin) {
      yield call(forwardTo, AFTER_LOGIN_REDIRECT.admin);
    }
  } catch (err) {
    yield put(loginError(err));
  }
}

// Individual exports for testing
export default function* defaultSaga() {
  yield takeLatest(LOGIN_REQUEST, login);
  yield takeLatest(FACEBOOK_LOGIN_REQUEST, facebookLogin);
}

/**
 * Helper to handle navigation from sagas.
 * @param  {Sting} location The path to navigate
 */
function forwardTo(location) {
  history.replace(location);
}
