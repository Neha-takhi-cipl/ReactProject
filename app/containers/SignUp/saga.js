/**
 * SignUp saga call api for signup container
 *
 */

import { all, call, takeLatest, select, put } from 'redux-saga/effects';
import { SIGNUP_REQUEST } from 'containers/App/constants';
import { FACEBOOK_SIGNUP } from 'containers/SignUp/constants';
import { signupSuccess, signupError, addUserToStoreSuccess } from 'containers/App/actions';
import request from 'utils/request';
import { history } from 'app';
import auth from 'utils/auth';
import { USER_TYPE, API_ENDPOINTS, AFTER_LOGIN_REDIRECT } from 'appConfig';
import { makeSelectFullname, makeSelectEmail, makeSelectPassword, makeSelectFacebooksignup } from 'containers/SignUp/selectors';

/**
 * @description SignUp the user  request/response handler
 * @author PravinKumar
 * @since 17 May 2018
 */
export function* signUp() {
  // Select username from store
  const fullname = yield select(makeSelectFullname());
  const email = yield select(makeSelectEmail());
  const password = yield select(makeSelectPassword());
  const requestURL = `${API_ENDPOINTS.register}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({
      fullName: fullname,
      email,
      password,
      facebookLogin: false,
      userType: USER_TYPE.traveller,
    }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    // Call our request helper (see 'utils/request')
    const Response = yield call(request, requestURL, requestParam);
    if (Response.status === 201) {
      yield put(signupSuccess(Response));
      yield put(addUserToStoreSuccess(Response));

      yield all([
        call(auth.setToken, Response.data.token, true),
        call(auth.setUserInfo, Response.data, true),
      ]);
      if (Response.data.userType === USER_TYPE.traveller) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.traveller);
      } else if (Response.data.userType === USER_TYPE.agent) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.agent);
      } else if (Response.data.userType === USER_TYPE.admin) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.admin);
      }
    } else {
      yield put(signupSuccess(Response));
      yield put(addUserToStoreSuccess(Response));
    }
  } catch (err) {
    yield put(signupError(err));
  }
}

/**
 * @description SignUp the user  request/response handler
 * @author PravinKumar
 * @since 18 May 2018
 */
export function* facebooSignup() {
  // Select username from store
  const facebooksignup = yield select(makeSelectFacebooksignup());
  const requestURL = `${API_ENDPOINTS.register}`;
  const requestParam = {
    method: 'POST',
    body: JSON.stringify({
      fullName: facebooksignup.name,
      email: facebooksignup.email,
      facebookId: facebooksignup.facebookId,
      picture: facebooksignup.picture,
      facebookLogin: true,
      password: false,
      userType: USER_TYPE.traveller,
    }),
    headers: {
      'Content-Type': 'application/json',
    },
  };
  try {
    // Call our request helper (see 'utils/request')
    const Response = yield call(request, requestURL, requestParam);
    if (Response.status === 201 || Response.status === 422) {
      yield put(signupSuccess(Response));
      yield put(addUserToStoreSuccess(Response));

      yield all([
        call(auth.setToken, Response.data.token, true),
        call(auth.setUserInfo, Response.data, true),
      ]);
      if (Response.data.userType === USER_TYPE.traveller) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.traveller);
      } else if (Response.data.userType === USER_TYPE.agent) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.agent);
      } else if (Response.data.userType === USER_TYPE.admin) {
        yield call(forwardTo, AFTER_LOGIN_REDIRECT.admin);
      }
    } else {
      yield put(signupError(Response));
    }
  } catch (err) {
    yield put(signupError(err));
  }
}

/**
 * @description Root saga manages watcher lifecycle
 * @author PravinKumar
 * @since 17 May 2018
 */

export default function* signUpWatcher() {
  // Watches for SIGNUP_REQUEST actions and calls signup when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(SIGNUP_REQUEST, signUp);
  yield takeLatest(FACEBOOK_SIGNUP, facebooSignup);
}

/**
 * Helper to handle navigation from sagas.
 * @param  {Sting} location The path to navigate
 */
function forwardTo(location) {
  history.replace(location);
}
