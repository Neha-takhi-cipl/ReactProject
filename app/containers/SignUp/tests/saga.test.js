/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */

// import { put, takeLatest } from 'redux-saga/effects';
// import { SIGNUP_REQUEST, BASE_URL } from 'containers/App/constants';
// import { signupSuccess, signupError } from 'containers/App/actions';

import { signUp } from '../saga';
describe('signUp Saga', () => {
  beforeEach(() => {
    const signUpGenerator = signUp();

    const selectDescriptor = signUpGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('Expect to have unit tests specified', () => {
    expect(true).toEqual(true);
  });
});
