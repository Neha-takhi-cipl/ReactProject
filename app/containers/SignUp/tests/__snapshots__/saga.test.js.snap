// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp Saga Expect to have unit tests specified 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
